### ISSUE TITLE
Lightweight Command Line Linux In The Gallery?

### ISSUE THREAD

**Issue by MartinPacker** [utmapp/UTM#6494]
Could we have a lightweight Linux image that can run in UTM SE eg on an iPad?

There are lots of things I can't do on iPad OS this would enable.

I don't need a GUI so a command line build would do fine.

I do need a proper package installation capability.

Also this would need to be save-able.

**Comment by lgrachov** [comment]
There is already an Arch Linux machine in the gallery and a Debian 10.4 Custom i3. Which one fits you the best? All of them are command line and the Debian 10.4 Custom i3 has a console.

**Comment by MartinPacker** [comment]
Thank you. The one that successfully got me to a command line was Arch Linux. However, trying to run `pacman -S python` got me a 404. In fact any `pacman -S` attempt to install anything gives me the same result.

I would've preferred the Debian route but I can't get to a command line when it starts.

**Comment by MartinPacker** [comment]
Actually fixed the pacman issue with `pacman -Syy` - to update its list of packages.

**Comment by tosigus** [comment]
Thanks for the pointer to Debian 10.4, it helps but some programs are working in my existing workflows better with Ubuntu. Would a minimal Ubuntu Server version also be possible?

**Comment by lgrachov** [comment]
> Thanks for the pointer to Debian 10.4, it helps but some programs are working in my existing workflows better with Ubuntu. Would a minimal Ubuntu Server version also be possible? 

Yes. If the `.iso` you have is Ubuntu Server 22.04, then there is an Ubuntu 22.04 and 14.04 guide in the UTM [Gallery](https://mac.getutm.app/gallery)

**Comment by lgrachov** [comment]
> I would've preferred the Debian route but I can't get to a command line when it starts.

I have read the documentation of Debian, the operating system prompts you to first choose whether you want to use <kbd>Option</kbd> or <kbd>Command</kbd> as the `$Mod` key. The `$Mod` key stands for the modifier key, which is the same thing as the  <kbd>Command</kbd> key. The <kbd>Command</kbd> can be used with the <kbd>Space</kbd> key to open the Spotlight menu. The prompt is shown only when the `config` file does not exist at `~/.i3/` or `~/.config/i3/`.

The virtual machine template has it set to <kbd>Command</kbd> or <kbd>Option</kbd>. Try pressing them to see which one is the modifier key. After the modifier key has been set you can press <kbd>Modifier+D</kbd> to open the application startup `dmenu`. With `dmenu` you can then open a terminal or text editor to read or modify the configuration file.

In the configuration file, you can add key binds to open applications with this format:
```bash
bindsym $Mod+<key> <command> <arguments>
```

So to open Firefox with the Debian site by pressing <kbd>Modifier+F</kbd>, you can do this:
```bash
bindsym $Mod+F exec firefox www.debian.org 
```

You will have to restart the window system for the new configuration file to take changes:
```bash
- $ i3 restart
```

> [!TIP]
> You can set the restart command to a key bind can save a lot of time during configuration:

```bash
bindsym $Mod+<key> restart
```

For more information about setting up i3, check [their documentation](https://i3wm.org/docs/).

Hope this helps!

**Comment by leszek-s** [comment]
I recently tried using antiX linux distro (I checked antiX-23.1-Base 32bit with lightweight gui) on my ipad in UTM SE and while it was still very slow I definitely worked better than other linux distros from utm gallery (and also better than reactos) so perhaps you might want to try that. I also wanted to try freedos but I had problems with setting it up. Perhaps this antiX (or something similar with very lightweight gui) or maybe freedos (if anyone is able to set it up) could be added to offical utm gallery so we have there something that works faster on iOS with UTM SE?

**Comment by lgrachov** [comment]
> or maybe freedos (if anyone can set it up)

This is a guide for macOS, but I hope you can figure it out.

Create a new virtual machine and choose **Emulation**. It is gonna be slow because we chose emulation. Choose your `.iso` file in the **Boot iso path**. Set about 100 megabytes to the main drive. That is enough to run DOS. Then, set anywhere about 512 megabytes of memory. Choose **Open VM settings** after creation.

Add a new drive. Set it to a floppy drive, and save all of the changes. Adjust the order of the drives so that the floppy is in the bottom, the CD-ROM is at the top, and the main hard drive is in the middle.

When booted, press <kbd>1</kbd> or <kbd>Enter</kbd>. Then, choose the run `fdisk` option. If it asks if you want to use FAT32, choose yes. Choose option 1 for the primary disk partition. Then, choose yes. (the partition you install FreeDOS onto must be active so it boots.)

Boot from the CD, and run `SETUP`. Format the drive if asked to, and continue the setup. I provided an existing UTM template so you don't have to do this yourself. I don't have a FreeDOS `.iso`, but I made the template on iPad mini 6.

[FreeDOS.zip](https://github.com/user-attachments/files/16429081/FreeDOS.zip)

**Comment by lgrachov** [comment]
@MartinPacker If you can make a lightweight Linux virtual machine, you can then ask to submit it to the gallery.

I found out that you can contribute to the gallery. It was explained in https://github.com/utmapp/UTM/issues/3807 .
